hasher.go

```diff
@@ -154,10 +154,12 @@ func (n *NmtHasher) BlockSize() int {
 
 func (n *NmtHasher) EmptyRoot() []byte {
 	n.baseHasher.Reset()
-	h := n.baseHasher.Sum(nil)
-	digest := append(make([]byte, int(n.NamespaceLen)*2), h...)
+	// make returns a zeroed slice, exactly what we need for the (nID || nID)
+	zeroSize := int(n.NamespaceLen) * 2
+	fullSize := zeroSize + n.baseHasher.Size()
 
-	return digest
+	digest := make([]byte, zeroSize, fullSize)
+	return n.baseHasher.Sum(digest)
```

Comment:
I wrote a unit test to verify this refactor doesn't modify the returned empty root: https://github.com/celestiaorg/nmt/pull/288

Didn't want to push to your branch without permission but feel free to include in this PR if it helps

Reply:
Done. Have included your test, thank you!